pkg/dist/buildmodule.go

```diff
@@ -59,6 +61,30 @@ type Target struct {
 	Distributions []Distribution `json:"distributions,omitempty" toml:"distributions,omitempty"`
 }
 
+// ValuesAsSlide converts the internal representation of a target, os, arch, variant, etc. into a string slide
+// each value included into the final array must be not empty.
+func (t *Target) ValuesAsSlide(version string) []string {
+	var targets []string
+	if t.OS != "" {
+		targets = append(targets, t.OS)
+	}
+	if t.Arch != "" {
+		targets = append(targets, t.Arch)
+	}
+	if t.ArchVariant != "" {
+		targets = append(targets, t.ArchVariant)
+	}
```

Comment:
Is it possible that `t.Arch` could be empty and `t.ArchVariant` is non-empty? Do we need to guard against that case?

Reply:
Interesting, according to the spec:
> variant string, OPTIONAL

>The variant of the specified CPU architecture. Configurations SHOULD use, and implementations SHOULD understand, variant values listed in the [Platform Variants](https://github.com/opencontainers/image-spec/blob/main/image-index.md#platform-variants) table.

it looks like you have to specify the architecture, right?

Reply:
Can we rely on the user input to be spec compliant?

Reply:
I think we should add some sort of validation here

Reply:
Maybe, checking for example [ggcr](https://github.com/google/go-containerregistry/blob/main/pkg/v1/platform.go), they actually doesn't do any sort of validation

Reply:
Ok